src/main/java/pro/sky/shoppingcart/Cart.java

```diff
@@ -0,0 +1,12 @@
+package pro.sky.shoppingcart;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.annotation.SessionScope;
+import java.util.List;
+@Component
+@SessionScope
+public class Cart {
+   public List<String> productsList;
+   public Cart(List<String> productsList) {this.productsList = productsList;}
```

Comment:
```suggestion
   public List<String> productsList = new ArrayList<>();
   public Cart() {}
```
Тут ижектить лист не нужно. Напиши вот так.